### ISSUE TITLE
[BUG] Pannels only float, impossible to drop and attach to structure 24.9.6 KDE FLATPAK ONLY wayland qt 5.15

### ISSUE THREAD

**Issue by joomlafun** [pbek/QOwnNotes#3109]
####System 24.9.6 KDE 5.27.8(kubuntu) wayland qt 5.15
!!Issue not present with the snap installation, only with the flatpak installation!!
#### Expected behaviour
Pannels to drop and attach to the main window when drag and drop is used

#### Actual behaviour
Pannels not dropping, they remain independent windows

#### Steps to reproduce
Unlock pannels
Pick a floating pannel, and drop it on the main structure

####
I saw the logs have a bit of personal content, please let me know if you need them for this situation
![Screenshot_20240915_1243](https://github.com/user-attachments/assets/36f048fe-b476-454f-a873-32f8c9d25d3b)

**Comment by pbek** [comment]
Hm, I cannot reproduce this. When I pull them out as a separate window, I'm able to drag them into the QOwnNotes main window again.

Hard to tell what's going on in your system. Did you try switching to another Layout in the layout system to bring them back?

**Comment by joomlafun** [comment]
Hi, 

Yes I tried to switch to another layout and did a reinstall.
It seems like the second window(the one with the panel to drop) is not aware of the main window of qownote.
Update : I've just installed the snap version and it WORKS fine...

However to have it solved : should I/you file a bug report be opened at flatpak and its compatibility with wayland? 
I cannot say at which update this issue started to happen I must say, I will stick with the snap.

**Comment by joomlafun** [comment]
I proposed to share the logs in my first post, it was a bit long to be honest and did not know that the copy paste removes the sensitive data
https://github.com/flathub/org.qownnotes.QOwnNotes/issues/295

**Comment by digisus** [comment]
Yes, that seems to be the same issue as mine #3135. My install on Debian Testing was via apt...

```
Version: 24.10.2-1
Download-Size: 4’476 kB
APT-Manual-Installed: yes
APT-Sources: http://download.opensuse.org/repositories/home:/pbek:/QOwnNotes/Debian_12  Packages
```

Could it be related to Wayland? I think I saw a warning related to it in the log...
BTW: I am actually not able to get the navigation panel back to where the others are. I can only switch it on/off.

**Comment by pbek** [comment]
I just tested again on NixOS with KDE Plasma and Wayland, and snapping the panel back still worked fine. 🤔

**Comment by digisus** [comment]
It would help as a dirty hack, if I would know in which config file the panel positions are stored, so I can hack around to get the navi panel docked again - then, I would just leave it there for the time being. Thanks.

**Comment by pbek** [comment]
The file path is shown in the Debug settings.

**Comment by pbek** [comment]
Today I was able to reproduce it on one machine (NixOS, Qt6, Wayland). But I don't know why I could reproduce it here only (I have nvidia here) and I don't know what I should do about it. Looks like a Qt issue. 🤔

**Comment by pbek** [comment]
## 25.2.9
- There now is a new action in the *Window menu* to reattach all floating panels in case
  they can't be reattached manually anymore (for [#3109](https://github.com/pbek/QOwnNotes/pull/3109))

There now is a new release, could you please test it and report if it works for you?

**Comment by joomlafun** [comment]
@pbek 
Hi, thank you for this feature, I wonder what QT version you are on for your test but on  my side since update to KDE 6.2.5 QT 6.8.1 on Fedora 41 I do not have the issue.
The provided feature works well and should help other users experiencing this situation.
@digisus ; are you still experiencing the situation after updating QT?

**Comment by pbek** [comment]
Qt 6.8.2, but I only had the issue with the Nvidia GPU, not on the system with the AMD GPU. ️🤷🏻

**Comment by digisus** [comment]
Thanks for looking into this again...

I am on QON v25.2.9 now (with Qt 5.15.5 on a Debian GNU/Linux trixie/sid). 

I still can not re-attached dettached panels, but I can see the new menu item under "Window" and it works to re-attach the panels! 

So, thanks for that. 👍

**Comment by tannenfreund87** [comment]
It's an upstream issue: https://qt-project.atlassian.net/browse/QTBUG-87332

On current Qt 6.10, the re-attaching works, but the panels don't survive a restart. We'll probably have to wait until this Wayland protocol get's approved by xdg and implemented by Qt and the major compositors: https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/264

**Comment by pbek** [comment]
Is this still an issue? I'll re-open if it is. 😊

**Comment by tannenfreund87** [comment]
The original issue is solved. At least on my machine:

System:
> Operating System: openSUSE Tumbleweed 20260610
> KDE Plasma Version: 6.6.5
> KDE Frameworks Version: 6.26.0
> Qt Version: 6.11.1
> Kernel Version: 7.0.11-1-default (64-bit)
> Graphics Platform: Wayland
> Processors: 12 × AMD Ryzen 5 7535HS with Radeon Graphics
> Memory: 32 GiB of RAM (30.1 GiB usable)
> Graphics Processor: AMD Radeon 660M

QGIS:
QGIS version | 4.0.3-Norrköping
-- | --
QGIS code revision | a60051e0f48
 
Libraries
Qt version | 6.11.1
Python version | 3.13.13
GDAL version | 3.13.1 — Iowa City
PROJ version | 9.8.1
EPSG Registry database version | v12.029 (2025-10-02)
GEOS version | 3.14.1-CAPI-1.20.5
SFCGAL version | No support
GeographicLib version | No support
SQLite version | 3.53.2
PDAL version | 2.10.1
PostgreSQL client version | 17.10
SpatiaLite version | 5.1.0
QWT version | 6.3.0
QScintilla2 version | 2.14.1
OS version | openSUSE Tumbleweed
This copy of QGIS writes debugging output.

I occasionally get a QGIS crash, if I undock the panels, but I cannot reproduce:
[QGIS_crash.txt](https://github.com/user-attachments/files/29287552/QGIS_crash.txt)
Docking and undocking works, the panels are also undocked after a restart, albeit scattered around and covered by the main application window. 

Generally, the floating panels are unusable under wayland, but this is a massive upstream issue. Wayland devs never thought about the need of having ONE app with multiple windows and docks. An established UI design in the CAD and GIS space for decades! And it doesn't look like it will get solved soon, so I'll have to switch desktops after plasma 6.8 :-(

**Comment by pbek** [comment]
Thank you for the report!
You get a qgis crash when you undock panels in QOwnNotes? How are those apps related? 🤔

> so I'll have to switch desktops after plasma 6.8

Isn't there already a fork of kde plasma for still using xorg?
And there also is yserver now. 😊

**Comment by tannenfreund87** [comment]
Sorry, I got the bug reports confused! Today at noon I already had 35° in my office.

QGIS and QOwnNotes both use the Qt docking system and QGIS has the same or similar problems. I'm also following the respective bugs/issues of QGIS: https://github.com/qgis/QGIS/issues/29177#issuecomment-3310828432

Compiled with Qt 6.11, the docks in QOwnNotes show the same behavior as in QGIS (but no crashes, yay!):

> Docking and undocking works, the panels are also undocked after a restart, albeit scattered around and covered by the main application window. Generally, the floating panels are unusable under wayland, but this is a massive upstream issue.

**Comment by pbek** [comment]
Ah, I see. Thank you. 😊
